### PR TITLE
First version

### PR DESCRIPTION
**Pull request by Krulknul**
First version of the client after a refactor to use a builder pattern for building requests.
There are a few, but insufficient tests at present. If this gets merged, I can write more tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,60 @@
...
+
+// Instantiate a new client with a base URL
+let client = GatewayClientAsync::new(
+    // or use radix_client::constants::PUBLIC_GATEWAY_URL
+    "https://mainnet.radixdlt.com".to_string(),
+);
+
+// Use a builder pattern to create and execute the request
+let response = client
+    .get_transactions_stream_builder()
```
Comment (fpieper) [inline README.md]: Normally by convention `get_` is not common in Rust.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,60 @@
...
+    .order(Order::Asc)
+    .limit_per_page(10)
```
Comment (fpieper) [inline README.md:51]: Does the builder pattern work well with querying multiple pages using the cursor?

Do we have an alternative to `execute`? Maybe `fetch` or something you would always need to decide like `json()` or `xml()` (I now we don't have xml ;) )?

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+impl client_type {
+    pub fn new(base_url: String) -> client_type {
+        client_type {
+            base_url,
+            client: reqwest_client_type::new(),
+        }
+    }
+
+    #[maybe_async_attr]
+    pub async fn post_request<S: Serialize>(
```
Comment (fpieper) [inline src/lib.rs]: I would probably name it `post`, but is mainly internal anyway 👍

File: src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
```
Comment (fpieper) [inline src/lib.rs]: Here too `post`

File: src/gateway/stream/get_transactions_stream.rs [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+                let mut opt_ins = TransactionStreamOptIns::default();
+                opt_ins.raw_hex = true;
+                Some(opt_ins)
+            }
+        };
+        self
+    }
+
+    pub fn with_receipt_state_changes(&mut self) -> &mut Self {
+        self.request.opt_ins = match self.request.opt_ins.clone() {
```
Comment (fpieper) [inline src/gateway/stream/get_transactions_stream.rs]: For the Opt-Ins there seems to be a lot of boilerplate? I guess the code can be reduced? Using `map`, and/or default on `Option`

**Review by fpieper — COMMENTED** [review]
Thank you very much. Don't have lots of feedback since it is looking good overall 👍 and the structure is mostly defined by the underlaying API

File: README.md (hunk 1 above)
Comment (Krulknul) [reply README.md]: let's remove that, but keep the builder suffix

File: src/gateway/stream/get_transactions_stream.rs (hunk 5 above)
Comment (Krulknul) [reply src/gateway/stream/get_transactions_stream.rs]: yes, will change this to be shorter

**Pull request merged**